develop/core/views.py

```diff
@@ -21,35 +20,33 @@ def get_queryset(self):
         return self.model.on_site.all()
 
 
-class ProductAccessView(ProductRequiredMixin, TemplateView):
+class ProductAccessView(SetSiteToRequestMixin, ProductRequiredMixin, TemplateView):
     model = Offer
     template_name = "core/product_use.html"
 
     def get(self, request, *args, **kwargs):
         context = super().get_context_data(**kwargs)
-        site = get_current_site(request)
-        context['object'] = self.model.objects.get(site=site, slug=kwargs['slug'])
+
+        context['object'] = self.model.objects.get(site=request.site, slug=kwargs['slug'])
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
changed to: `site=get_site_from_request(request)`